Tone/component/filter/Filter.ts

```diff
@@ -0,0 +1,76 @@
+import {ToneAudioNode, ToneAudioNodeOptions, InputNode, OutputNode} from '../../core/context/ToneAudioNode';
+import { optionsFromArguments } from "Tone/core/util/Defaults";
+import { FrequencyClass } from 'Tone/core/type/Frequency';
+
+type FilterType = "lowpass" | "highpass" | "bandpass" | "lowshelf" | "highshelf" | "peaking" | "notch" | "allpass";
+
+interface FilterOptions extends ToneAudioNodeOptions {
+	type: FilterType;
+	frequency: number;
+	rolloff: number;
+	Q: number;
+	gain: number
+}
+
+
+/**
+ *  @class  Tone.Filter is a filter which allows for all of the same native methods
+ *          as the [BiquadFilterNode](http://webaudio.github.io/web-audio-api/#the-biquadfilternode-interface).
+ *          Tone.Filter has the added ability to set the filter rolloff at -12
+ *          (default), -24 and -48.
+ *
+ *  @constructor
+ *  @extends {Tone.AudioNode}
+ *  @param {Frequency|Object} [frequency] The cutoff frequency of the filter.
+ *  @param {string=} type The type of filter.
+ *  @param {number=} rolloff The drop in decibels per octave after the cutoff frequency.
+ *                            3 choices: -12, -24, and -48
+ *  @example
+ *  var filter = new Tone.Filter(200, "highpass");
+ */
+export class Filter extends ToneAudioNode<FilterOptions> {
+
+	name = "Filter";
+	input = InputNode;
+
+	constructor(type?: FilterOptions);
+	constructor(options?: Partial<FilterOptions>);
+	constructor() {
+		super(optionsFromArguments(Filter.getDefaults(), arguments, ["frequency", "type", "rolloff"]));
+
+		this.createInsOuts(1,1);
```

Comment:
i think no need for this function. check out [Analyser.ts](https://github.com/Tonejs/Tone.js/blob/typescript/Tone/component/analysis/Analyser.ts) which has a similar thin wrapping of a native node. it just declares the filter as the input and output node.